src/main/java/org/radarcns/management/web/rest/AuditResource.java

```diff
@@ -37,6 +46,7 @@ public AuditResource(AuditEventService auditEventService) {
      */
     @GetMapping
     public ResponseEntity<List<AuditEvent>> getAll(@ApiParam Pageable pageable) {
+        checkPermission(getJWT(servletRequest), AUDIT_READ);
```

Comment:
It would be nice to have declarative validation, i.e. something like
```
@GetMapping
@RequiresPermission(AUDIT_READ)
public ResponseEntity<List<AuditEvent>> getAll(@ApiParam Pageable pageable) {
```
but the current implementation is also ok

Reply:
Indeed, that was my first idea was well. For simple permissions such as this we could probably make something like that work. The challenge comes from the fact that other permissions are scoped to a project, e.g. a project analyst in one project can read subjects there, but not the subjects of another project. I'm keeping in the back of my mind that we can move to Spring ACLs in the future as well if we need more expressiveness.